packages/typespec-azure-resource-manager/lib/operations.tsp

```diff
@@ -50,14 +50,24 @@ op ArmResourceListByParent<
   ParentName extends valueof string = "",
   ParentFriendlyName extends valueof string = "",
   Parameters extends {} = {},
-  Response extends {} = ArmResponse<ResourceListResult<Resource>>,
+  Response = ArmResponse<ResourceListResult<Resource>>,
   Error extends {} = ErrorResponse
-> is ArmReadOperation<
+> is ArmCustomReadOperation<
   ResourceParentParameters<Resource, BaseParameters> & Parameters,
   Response,
   Error
 >;
 
+/**
+ * @dev Template for Azure Resource Manager GET and HEAD Operations.
+ * @param Parameters The parameter object for the operation.
+ * @param Response The response or union of responses for success, it can be any type.
+ * @param ErrorResponse The error response.
+ */
+op ArmCustomReadOperation<Parameters extends {}, Response, ErrorResponse extends {}>(
```

Comment:
there is a conflict with the response type on the `ArmReadOperation`:
`Argument of type 'Response' is not assignable to parameter of type '{}'`

Reply:
Oh, I see.  This is because the PR removes the constraint on `Response` people should not use this template unless they are returning a multi-page response, meanign that the response will always be a model.  The `items` property in the response may not be an array of models, but that is not what this parameter controls.

Reply:
thanks a lot for the explanation, I worked on this, I have another question related to this item, should I also relax the Resource part of:   https://github.com/Azure/typespec-azure/blob/8225bab1043fc4b1cf87790eb457f12a94420634/packages/typespec-azure-resource-manager/lib/operations.tsp#L23

so it can be any model not only foundation.resource?  as that was mentioned originally in the issue: https://github.com/Azure/typespec-azure/issues/262 :)

Reply:
No, I don't think we should loosen the restriction on the operation, I think this issue was  slightly miswritten - the operation is still a resource list operation, which means it needs to be an operation over a resource, and the resource parameter determines the path, but only determines the default for the Response.  We should not have operations using the @armResourceList decorator that do not operate over an ARM resource.

Reply:
perfect! thanks Mark!